modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -283,11 +334,17 @@ public GridEncryptionManager(GridKernalContext ctx) {
 
         performMKChangeProc = new DistributedProcess<>(ctx, MASTER_KEY_CHANGE_FINISH, this::performMasterKeyChange,
             this::finishPerformMasterKeyChange);
+
+        reencryption = new CacheGroupReencryption(ctx);
```

Comment:
could you put this dependency outside?

Reply:
Could you describe in more detail what should we do here?